src/main/java/org/owasp/benchmark/score/parsers/HdivReader.java

```diff
@@ -0,0 +1,162 @@
+package org.owasp.benchmark.score.parsers;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.owasp.benchmark.score.BenchmarkScore;
+
+public class HdivReader extends Reader {
+
+	Set<String> invalid = new HashSet<>();
+
+	public static void main(final String[] args) throws Exception {
+		File f = new File("hdivAgentLog.hlg");
+		HdivReader cr = new HdivReader();
+		System.out.println(cr.parse(f));
+	}
+
+	public TestResults parse(final File f) throws Exception {
+		TestResults tr = new TestResults("Hdiv", true, TestResults.ToolType.IAST);
+
+		BufferedReader reader = new BufferedReader(new FileReader(f));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *reader*. The resource is referenced in statements at the following lines: 30, 31, 38. The resource closure statement is at line: 62. There are other execution paths that do not contain closure statements, e.g., when exception is thrown by BufferedReader.readLine. Either a) close *reader* in a try-finally block or b) close the resource by declaring *reader* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).